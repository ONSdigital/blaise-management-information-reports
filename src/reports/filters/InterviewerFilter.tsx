import {StyledForm} from "blaise-design-system-react-components";
import React, {ReactElement} from "react";
import dateFormatter from "dayjs";
import Breadcrumbs from "../../components/Breadcrumbs";
import CallHistoryLastUpdatedStatus from "../../components/CallHistoryLastUpdatedStatus";

interface InterviewerFilterPageProps {
    title: string
    interviewer: string | undefined
    setInterviewer: (string: string) => void
    startDate: Date
    setStartDate: (Date: Date) => void
    endDate: Date
    setEndDate: (Date: Date) => void
    surveyTla: string | undefined
    setSurveyTla: (string: string) => void
    submitFunction: () => void;
}

function InterviewerFilter(props: InterviewerFilterPageProps): ReactElement {
    const {
        title,
        interviewer,
        setInterviewer,
        startDate,
        setStartDate,
        endDate,
        setEndDate,
        surveyTla,
        setSurveyTla,
        submitFunction
    } = props;

    async function submitInterviewerFilters(formValues: Record<string, any>, setSubmitting: (isSubmitting: boolean) => void): Promise<void> {
        setInterviewer(formValues["Interviewer ID"]);
        setStartDate(formValues["Start date"]);
        setEndDate(formValues["End date"]);
        setSurveyTla(formValues["Survey TLA"]);
        setSubmitting(true);
        submitFunction();
    }

    const validateInterviewer = (value: string) => {
        let error;
        if (value === "" || value === undefined) {
            error = "Enter a interviewer ID";
        }
        return error;
    };

    const validateDate = (value: string) => {
        let error;
        if (value === "" || value === undefined) {
            error = "Enter a date";
            return error;
        }

        if (value.match(/^[0-9]{4}-[0-9]{2}-[0-9]{2}$/) === null) {
            error = "Enter a valid date";
        }
        return error;
    };

    const fields = [
        {
            name: "Survey TLA",
            description: "Select survey",
            type: "radio",
            initial_value: surveyTla,
            radioOptions: [
                {id: "all", value: "", label: "Show all surveys"},
                {id: "lms", value: "lms", label: "LMS", description: "Labour Market Survey"},
                {id: "opn", value: "opn", label: "OPN", description: "Opinions and Lifestyle Survey"}
            ]
        },
        {
            name: "Interviewer ID",
            type: "text",
            initial_value: interviewer,
            validate: validateInterviewer
        },
        {
            name: "Start date",
            type: "date",
            initial_value: dateFormatter(startDate).format("YYYY-MM-DD"),
            validate: validateDate
        },
        {
            name: "End date",
            type: "date",
            initial_value: dateFormatter(endDate).format("YYYY-MM-DD"),
            validate: validateDate
        }
    ];

    return (
        <>
            <div>
                <Breadcrumbs
                    BreadcrumbList={[{link: "/", title: "Reports"}]}/>
                <main id="main-content" className="page__main u-mt-s">
                    <h1 className="u-mb-m">Run interviewer {title} report</h1>
                    <CallHistoryLastUpdatedStatus/>
                    <StyledForm fields={fields} onSubmitFunction={submitInterviewerFilters} submitLabel={"Next"}/>
                </main>
            </div>
        </>
    );
}

export default InterviewerFilter;


