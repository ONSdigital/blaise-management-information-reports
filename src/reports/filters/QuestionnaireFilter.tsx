import React, { ReactElement } from "react";
import Breadcrumbs from "../../components/Breadcrumbs";
import CallHistoryLastUpdatedStatus from "../../components/CallHistoryLastUpdatedStatus";
import QuestionnaireSelector from "../../components/QuestionnaireSelector";
import { bstDateFormatter } from "../../utilities/Helpers";

interface QuestionnaireFilterPageProps {
    interviewer: string
    startDate: Date
    endDate: Date
    surveyTla: string
    questionnaires: string[]
    setQuestionnaires: (string: string[]) => void
    submitFunction: () => void
    navigateBack: () => void
}

function QuestionnaireFilter(props: QuestionnaireFilterPageProps): ReactElement {
    const {
        interviewer,
        startDate,
        endDate,
        navigateBack,
    } = props;

    return (
        <>
            <div>
                <Breadcrumbs
                    BreadcrumbList={[{ link: "/", title: "Reports" }, {
                        link: "#",
                        onClickFunction: navigateBack,
                        title: "Interviewer details"
                    }]}/>
                <main id="main-content" className="page__main u-mt-s">
                    <h1>Select questionnaires for</h1>
                    <h3 className="u-mb-m">
                        Interviewer: {interviewer}<br></br>
                        Period: {bstDateFormatter(startDate)}â€“{bstDateFormatter(endDate)}
                    </h3>
                    <CallHistoryLastUpdatedStatus/>
                    <QuestionnaireSelector {...props}/>
                </main>
            </div>
        </>
    );
}

export default QuestionnaireFilter;
