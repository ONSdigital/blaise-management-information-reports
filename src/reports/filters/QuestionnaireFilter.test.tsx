/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";
import { createMemoryHistory, History } from "history";
import { render, waitFor } from "@testing-library/react";
import { Router } from "react-router";
import { act } from "react-dom/test-utils";
import { fireEvent, screen } from "@testing-library/dom";
import React from "react";
import QuestionnaireFilter from "./QuestionnaireFilter";
import MockAdapter from "axios-mock-adapter";
import axios from "axios";
import dateFormatter from "dayjs";
import utc from "dayjs/plugin/utc";
import timezone from "dayjs/plugin/timezone";
import subtractYears from "../../utilities/DateFormatter";
import { InterviewerFilterQuery } from "./InterviewerFilter";

const mockAdapter = new MockAdapter(axios);

const questionnaireDataReturned: string[] = [
    "LMS2101_AA1",
    "LMS2101_AA2",
];

dateFormatter.extend(utc);
dateFormatter.extend(timezone);

describe("the interviewer details page renders correctly", () => {
    let history: History;
    let setQuestionnaires: (questionnaires: string[]) => void;
    let submit: () => void;

    beforeEach(() => {
        mockAdapter
            .onGet("/api/reports/call-history-status")
            //.reply(200, {last_updated: "Sat, 01 Jan 2000 10:00:00 GMT"});
            .reply(200, { last_updated: subtractYears(1) });
        history = createMemoryHistory();
        setQuestionnaires = jest.fn();
        submit = jest.fn();
    });

    afterEach(() => {
        mockAdapter.reset();
    });

    function renderComponent() {
        const interviewerFilterQuery: InterviewerFilterQuery = {
            interviewer: "James",
            startDate: new Date("2022-01-01"),
            endDate: new Date("2022-01-05"),
            surveyTla: "LMS"
        };

        return render(
            <Router history={ history }>
                <QuestionnaireFilter
                    interviewerFilterQuery={interviewerFilterQuery}
                    questionnaires={ ["LMS2101_AA1"] } setQuestionnaires={ setQuestionnaires }
                    onSubmit={ submit }
                    navigateBack={ () => {
                        return;
                    }}
                />
            </Router>
        );
    }

    it("posts the query params", async () => {
        expect.assertions(4);
        mockAdapter.onPost(
            "/api/questionnaires",
            {
                asymmetricMatch: (formData: FormData) => {
                    expect(formData.get("survey_tla")).toBe("LMS");
                    expect(formData.get("interviewer")).toBe("James");
                    expect(formData.get("start_date")).toBe("2022-01-01");
                    expect(formData.get("end_date")).toBe("2022-01-05");
                    return true;
                }
            }
        ).reply(200, questionnaireDataReturned);
        renderComponent();

        // Wait for loading to finish (avoid warnings)
        await screen.findByText("LMS2101_AA1");
    });

    it("matches loading snapshot", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();

        // Wait for loading to finish (avoid warnings)
        await screen.findByText("LMS2101_AA1");
    });

    it("matches snapshot", async () => {
        // This snapshot will need to be updated in 1 years time (28/06/2023)
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);
        mockAdapter
            .onGet("/api/reports/call-history-status")
            .reply(200, { last_updated: "Sat, 01 Jan 2000 10:00:00 GMT" });
        const wrapper = renderComponent();
        await screen.findByText("LMS2101_AA1");
        expect(wrapper).toMatchSnapshot();
    });

    it("renders correctly", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);

        await act(async () => {
            renderComponent();
        });

        expect(screen.queryByText(/Select questionnaire/i)).toBeVisible();
        expect(screen.queryByText(/Data in this report was last updated:/i)).toBeVisible();

        expect(screen.queryByText(/1 year ago/i)).toBeVisible();

        expect(screen.queryByText(/Interviewer: James/i)).toBeVisible();
        expect(screen.queryByText(/Period: 01\/01\/2022â€“05\/01\/2022/i)).toBeVisible();

        expect(screen.queryByText(/Questionnaire: LMS2101_AA1/i)).toBeVisible();
        expect(screen.queryByText(/Run report/i)).toBeVisible();
    });

    it("displays a message when not questionnaires are returned", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, []);
        renderComponent();
        await screen.findByText("No questionnaires found for given parameters.");
    });

    it("displays an error when an error HTTP status is returned", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(500, []);
        renderComponent();
        await screen.findByText("An error occurred while fetching the list of questionnaires");
    });

    it("displays an error when a non-200 success HTTP status is returned", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(201, questionnaireDataReturned);
        renderComponent();
        await screen.findByText("An error occurred while fetching the list of questionnaires");
    });

    it("checks all provided questionnaires by default", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);
        renderComponent();
        await act(async () => {
            fireEvent.click(await screen.findByText(/Run report/));
        });
        expect(setQuestionnaires).toHaveBeenCalledWith(["LMS2101_AA1"]);
        expect(submit).toHaveBeenCalled();
    });

    it("returns the questionnaires when a checkbox is ticket", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);
        renderComponent();
        await act(async () => {
            fireEvent.click(await screen.findByLabelText(/LMS2101_AA1/));
            fireEvent.click(await screen.findByLabelText(/LMS2101_AA2/));
            fireEvent.click(await screen.findByText(/Run report/));
        });
        expect(setQuestionnaires).toHaveBeenCalledWith(["LMS2101_AA2"]);
        expect(submit).toHaveBeenCalled();
    });

    it("displays an error when submitting with no checkboxes selected", async () => {
        mockAdapter.onPost("/api/questionnaires").reply(200, questionnaireDataReturned);
        renderComponent();
        await act(async () => {
            fireEvent.click(await screen.findByLabelText(/LMS2101_AA1/));
            fireEvent.click(await screen.findByText(/Run report/));
        });
        const elements = await screen.findAllByText("At least one questionnaire must be selected");
        expect(elements[0]).toBeVisible();
        expect(elements[1]).toBeVisible();
        expect(setQuestionnaires).not.toHaveBeenCalled();
        expect(submit).not.toHaveBeenCalled();
    });
});
