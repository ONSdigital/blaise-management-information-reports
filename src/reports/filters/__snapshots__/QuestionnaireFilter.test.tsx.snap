// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the interviewer details page renders correctly matches loading snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
        >
          <ol
            class="breadcrumb__items u-fs-s"
          >
            <li
              class="breadcrumb__item"
              id="breadcrumb-0"
            >
              <a
                class="breadcrumb__link"
                href="/"
              >
                Reports
              </a>
              <svg
                class="svg-icon"
                focusable="false"
                viewBox="0 0 8 13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                  transform="translate(-5.02 -1.59)"
                />
              </svg>
            </li>
            <li
              class="breadcrumb__item"
              id="breadcrumb-1"
            >
              <a
                class="breadcrumb__link"
                href="/"
              >
                Interviewer details
              </a>
              <svg
                class="svg-icon"
                focusable="false"
                viewBox="0 0 8 13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                  transform="translate(-5.02 -1.59)"
                />
              </svg>
            </li>
          </ol>
        </nav>
        <main
          class="page__main u-mt-s"
          id="main-content"
        >
          <h1>
            Select questionnaires for
          </h1>
          <h3
            class="u-mb-m"
          >
            Interviewer: 
            James
            <br />
            Period: 
            01/01/2022
            –
            05/01/2022
            <br />
            Questionnaire:
             
            LMS2101_AA1
          </h3>
          <p
            aria-live="polite"
            class="u-fs-s u-mt-s"
          >
            Data in this report was last updated: 
            <b>
              Loading
            </b>
          </p>
          <p
            class="u-fs-s"
          >
            Data in this report only goes back to the last 12 months.
          </p>
          <div
            class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
          >
            <div
              class="ons-panel__body "
            >
              <div
                style="text-align: center;"
              >
                <div
                  aria-busy="true"
                  class=""
                >
                  <svg
                    aria-label="audio-loading"
                    height="30"
                    viewBox="0 0 38 38"
                    width="30"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <lineargradient
                        id="a"
                        x1="8.042%"
                        x2="65.682%"
                        y1="0%"
                        y2="23.865%"
                      >
                        <stop
                          offset="0%"
                          stop-color="#064868"
                          stop-opacity="0"
                        />
                        <stop
                          offset="63.146%"
                          stop-color="#064868"
                          stop-opacity=".631"
                        />
                        <stop
                          offset="100%"
                          stop-color="#064868"
                        />
                      </lineargradient>
                    </defs>
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <g
                        transform="translate(1 1)"
                      >
                        <path
                          d="M36 18c0-9.94-8.06-18-18-18"
                          id="Oval-2"
                          stroke="#064868"
                          stroke-width="2"
                        >
                          <animatetransform
                            attributeName="transform"
                            dur="0.9s"
                            from="0 18 18"
                            repeatCount="indefinite"
                            to="360 18 18"
                            type="rotate"
                          />
                        </path>
                        <circle
                          cx="36"
                          cy="18"
                          fill="#fff"
                          r="1"
                        >
                          <animatetransform
                            attributeName="transform"
                            dur="0.9s"
                            from="0 18 18"
                            repeatCount="indefinite"
                            to="360 18 18"
                            type="rotate"
                          />
                        </circle>
                      </g>
                    </g>
                  </svg>
                </div>
                Loading
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
      >
        <ol
          class="breadcrumb__items u-fs-s"
        >
          <li
            class="breadcrumb__item"
            id="breadcrumb-0"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Reports
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
          <li
            class="breadcrumb__item"
            id="breadcrumb-1"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Interviewer details
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
        </ol>
      </nav>
      <main
        class="page__main u-mt-s"
        id="main-content"
      >
        <h1>
          Select questionnaires for
        </h1>
        <h3
          class="u-mb-m"
        >
          Interviewer: 
          James
          <br />
          Period: 
          01/01/2022
          –
          05/01/2022
          <br />
          Questionnaire:
           
          LMS2101_AA1
        </h3>
        <p
          aria-live="polite"
          class="u-fs-s u-mt-s"
        >
          Data in this report was last updated: 
          <b>
            Loading
          </b>
        </p>
        <p
          class="u-fs-s"
        >
          Data in this report only goes back to the last 12 months.
        </p>
        <div
          class="ons-panel ons-panel--info ons-panel--no-title  ons-u-mt-m"
        >
          <div
            class="ons-panel__body "
          >
            <div
              style="text-align: center;"
            >
              <div
                aria-busy="true"
                class=""
              >
                <svg
                  aria-label="audio-loading"
                  height="30"
                  viewBox="0 0 38 38"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <lineargradient
                      id="a"
                      x1="8.042%"
                      x2="65.682%"
                      y1="0%"
                      y2="23.865%"
                    >
                      <stop
                        offset="0%"
                        stop-color="#064868"
                        stop-opacity="0"
                      />
                      <stop
                        offset="63.146%"
                        stop-color="#064868"
                        stop-opacity=".631"
                      />
                      <stop
                        offset="100%"
                        stop-color="#064868"
                      />
                    </lineargradient>
                  </defs>
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <g
                      transform="translate(1 1)"
                    >
                      <path
                        d="M36 18c0-9.94-8.06-18-18-18"
                        id="Oval-2"
                        stroke="#064868"
                        stroke-width="2"
                      >
                        <animatetransform
                          attributeName="transform"
                          dur="0.9s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </path>
                      <circle
                        cx="36"
                        cy="18"
                        fill="#fff"
                        r="1"
                      >
                        <animatetransform
                          attributeName="transform"
                          dur="0.9s"
                          from="0 18 18"
                          repeatCount="indefinite"
                          to="360 18 18"
                          type="rotate"
                        />
                      </circle>
                    </g>
                  </g>
                </svg>
              </div>
              Loading
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`the interviewer details page renders correctly matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          aria-label="Breadcrumb"
          class="breadcrumb"
        >
          <ol
            class="breadcrumb__items u-fs-s"
          >
            <li
              class="breadcrumb__item"
              id="breadcrumb-0"
            >
              <a
                class="breadcrumb__link"
                href="/"
              >
                Reports
              </a>
              <svg
                class="svg-icon"
                focusable="false"
                viewBox="0 0 8 13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                  transform="translate(-5.02 -1.59)"
                />
              </svg>
            </li>
            <li
              class="breadcrumb__item"
              id="breadcrumb-1"
            >
              <a
                class="breadcrumb__link"
                href="/"
              >
                Interviewer details
              </a>
              <svg
                class="svg-icon"
                focusable="false"
                viewBox="0 0 8 13"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                  transform="translate(-5.02 -1.59)"
                />
              </svg>
            </li>
          </ol>
        </nav>
        <main
          class="page__main u-mt-s"
          id="main-content"
        >
          <h1>
            Select questionnaires for
          </h1>
          <h3
            class="u-mb-m"
          >
            Interviewer: 
            James
            <br />
            Period: 
            01/01/2022
            –
            05/01/2022
            <br />
            Questionnaire:
             
            LMS2101_AA1
          </h3>
          <p
            aria-live="polite"
            class="u-fs-s u-mt-s"
          >
            Data in this report was last updated: 
            <b>
              <time
                datetime="2000-01-01T10:00:00.000Z"
                title="2000-01-01 10:00"
              >
                23 years ago
              </time>
               (01/01/2000 10:00:00)
            </b>
          </p>
          <p
            class="u-fs-s"
          >
            Data in this report only goes back to the last 12 months.
          </p>
          <div
            class="input-items"
          >
            <div
              class="checkboxes__items"
            >
              <form
                action="#"
              >
                <fieldset
                  class="ons-fieldset"
                >
                  <legend
                    class="ons-fieldset__legend"
                  />
                  <button
                    class="ons-btn ons-u-mb-s js-auto-selector ons-btn--small ons-btn--secondary"
                    type="button"
                  >
                    <span
                      class="ons-btn__inner"
                    >
                      <span
                        class="js-button-text"
                      >
                        Select All
                      </span>
                      <span
                        class="ons-u-vh"
                      >
                         following checkboxes
                      </span>
                    </span>
                  </button>
                  <div
                    class="checkboxes__items"
                    id="questionnaires"
                  >
                    <p
                      class="ons-checkboxes__item"
                    >
                      <span
                        class="ons-checkbox"
                      >
                        <input
                          checked=""
                          class="ons-checkbox__input js-checkbox"
                          id="LMS2101_AA1"
                          initial_value="LMS2101_AA1"
                          name="questionnaires"
                          type="checkbox"
                          value="LMS2101_AA1"
                        />
                        <label
                          class="ons-checkbox__label "
                          for="LMS2101_AA1"
                          id="LMS2101_AA1-label"
                        >
                          LMS2101_AA1
                        </label>
                      </span>
                    </p>
                    <br />
                    <p
                      class="ons-checkboxes__item"
                    >
                      <span
                        class="ons-checkbox"
                      >
                        <input
                          class="ons-checkbox__input js-checkbox"
                          id="LMS2101_AA2"
                          initial_value="LMS2101_AA1"
                          name="questionnaires"
                          type="checkbox"
                          value="LMS2101_AA2"
                        />
                        <label
                          class="ons-checkbox__label "
                          for="LMS2101_AA2"
                          id="LMS2101_AA2-label"
                        >
                          LMS2101_AA2
                        </label>
                      </span>
                    </p>
                    <br />
                  </div>
                </fieldset>
                <br />
                <button
                  class="ons-btn "
                  data-testid="submit-button"
                  type="submit"
                >
                  <span
                    class="ons-btn__inner"
                  >
                    Run report
                  </span>
                </button>
              </form>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        aria-label="Breadcrumb"
        class="breadcrumb"
      >
        <ol
          class="breadcrumb__items u-fs-s"
        >
          <li
            class="breadcrumb__item"
            id="breadcrumb-0"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Reports
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
          <li
            class="breadcrumb__item"
            id="breadcrumb-1"
          >
            <a
              class="breadcrumb__link"
              href="/"
            >
              Interviewer details
            </a>
            <svg
              class="svg-icon"
              focusable="false"
              viewBox="0 0 8 13"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.74,14.28l-.57-.56a.5.5,0,0,1,0-.71h0l5-5-5-5a.5.5,0,0,1,0-.71h0l.57-.56a.5.5,0,0,1,.71,0h0l5.93,5.93a.5.5,0,0,1,0,.7L6.45,14.28a.5.5,0,0,1-.71,0Z"
                transform="translate(-5.02 -1.59)"
              />
            </svg>
          </li>
        </ol>
      </nav>
      <main
        class="page__main u-mt-s"
        id="main-content"
      >
        <h1>
          Select questionnaires for
        </h1>
        <h3
          class="u-mb-m"
        >
          Interviewer: 
          James
          <br />
          Period: 
          01/01/2022
          –
          05/01/2022
          <br />
          Questionnaire:
           
          LMS2101_AA1
        </h3>
        <p
          aria-live="polite"
          class="u-fs-s u-mt-s"
        >
          Data in this report was last updated: 
          <b>
            <time
              datetime="2000-01-01T10:00:00.000Z"
              title="2000-01-01 10:00"
            >
              23 years ago
            </time>
             (01/01/2000 10:00:00)
          </b>
        </p>
        <p
          class="u-fs-s"
        >
          Data in this report only goes back to the last 12 months.
        </p>
        <div
          class="input-items"
        >
          <div
            class="checkboxes__items"
          >
            <form
              action="#"
            >
              <fieldset
                class="ons-fieldset"
              >
                <legend
                  class="ons-fieldset__legend"
                />
                <button
                  class="ons-btn ons-u-mb-s js-auto-selector ons-btn--small ons-btn--secondary"
                  type="button"
                >
                  <span
                    class="ons-btn__inner"
                  >
                    <span
                      class="js-button-text"
                    >
                      Select All
                    </span>
                    <span
                      class="ons-u-vh"
                    >
                       following checkboxes
                    </span>
                  </span>
                </button>
                <div
                  class="checkboxes__items"
                  id="questionnaires"
                >
                  <p
                    class="ons-checkboxes__item"
                  >
                    <span
                      class="ons-checkbox"
                    >
                      <input
                        checked=""
                        class="ons-checkbox__input js-checkbox"
                        id="LMS2101_AA1"
                        initial_value="LMS2101_AA1"
                        name="questionnaires"
                        type="checkbox"
                        value="LMS2101_AA1"
                      />
                      <label
                        class="ons-checkbox__label "
                        for="LMS2101_AA1"
                        id="LMS2101_AA1-label"
                      >
                        LMS2101_AA1
                      </label>
                    </span>
                  </p>
                  <br />
                  <p
                    class="ons-checkboxes__item"
                  >
                    <span
                      class="ons-checkbox"
                    >
                      <input
                        class="ons-checkbox__input js-checkbox"
                        id="LMS2101_AA2"
                        initial_value="LMS2101_AA1"
                        name="questionnaires"
                        type="checkbox"
                        value="LMS2101_AA2"
                      />
                      <label
                        class="ons-checkbox__label "
                        for="LMS2101_AA2"
                        id="LMS2101_AA2-label"
                      >
                        LMS2101_AA2
                      </label>
                    </span>
                  </p>
                  <br />
                </div>
              </fieldset>
              <br />
              <button
                class="ons-btn "
                data-testid="submit-button"
                type="submit"
              >
                <span
                  class="ons-btn__inner"
                >
                  Run report
                </span>
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
